### ISSUE TITLE
Should Phoenix.Endpoint.RenderErrors log an error on uncaught :exit or :throw?

### ISSUE THREAD

**Issue by aselder** [phoenixframework/phoenix#4099]
### Environment

* Elixir version (elixir -v): 1.11.2/erlang 23.1.4
* Phoenix version (mix deps): 1.5.7
* NodeJS version (node -v):
* NPM version (npm -v):
* Operating system: OS X 1.10.6

### Expected behavior

If an uncaught exit or throw bubbles up to Phoenix.Endpoint.RenderErrors, I would expect a 500 status and a log message indicating what happened (probably at the level of :error)

### Actual behavior
The 500 is returned the client, but unless you explicitly write a telemetry handler for [:phoenix, :error_rendered] to do your own error handling, there is no indication in the logs as to what happened. Following the the Principle of Least Surprise, I think a log message at the error level would be expected.

I'm more that happy to whip up a PR if people agree with me.

**Comment by josevalim** [comment]
I cannot reproduce it. I see that everything is logged by default:

```
[error] #PID<0.673.0> running DemoWeb.Endpoint (connection #PID<0.658.0>, stream id 3) terminated
Server: localhost:4000 (http)
Request: GET /
** (exit) exited in: GenServer.call(:omg, :call, 5000)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started

```

Given you also filled an issue on Sentry, here is what is happening:

Failed requests are logged by Cowboy. However, Sentry does not publish log messages sent by Cowboy because it has its own plug. But, due to [the Sentry bug you filled](https://github.com/getsentry/sentry-elixir/issues/446), exit/throw are not being reported, which behaves as if we were not logging them (but we are). :)

So fixing this in Sentry should be enough. Alternatively, you can configure Sentry logger backend to not ignore messages coming from Cowboy, then you will also get throw/exit (but other exceptions will be logged twice).

Thanks for the report!

**Comment by aselder** [comment]
@josevalim We just ran into this situation tonight. Inside one of our actions, a call to Broadway.push_messages was causing the GenServer.call to exit; the pipeline name didn't exist.

Yet when we looked in our logs in DataDog there was no mention of the exits. Perhaps the exit message is being written directly to stdout or stderr and not using the Elixir logger.

**Comment by josevalim** [comment]
@aselder I just tried this quickly with Broadway and I was able to see it going through Logger. The way the message is formatted is very Logger specific (I added some Logger metadata to be double sure) and from looking at the code I can see it is using Logger too: https://github.com/elixir-plug/plug_cowboy/blob/master/lib/plug/cowboy/translator.ex

So I am afraid the message is being dropped somewhere. How are you pushing data to DataDog?
